lib/catalog/rbac/access.rb

```diff
@@ -14,55 +14,48 @@ def read_access_check
         resource_check('read')
       end
 
-      def create_access_check
-        permission_check('create')
+      def create_access_check(klass)
+        permission_check('create', klass)
       end
 
       def destroy_access_check
         resource_check('delete')
       end
 
-      def admin_check
-        return true unless rbac_enabled?
-
-        catalog_admin?
-      end
-
       def resource_check(verb, id = @record.id, klass = @record.class)
         return true unless rbac_enabled?
-        return true if catalog_admin?
 
-        return false unless access_object(@record.class.table_name, verb).accessible?
-        ids = access_id_list(verb, klass)
-        return false if klass.try(:supports_access_control?) && ids.exclude?(id.to_s)
+        scopes = access_object.scopes(@record.class.table_name, verb)
+        check = if scopes.include?("admin")
+          true
+        elsif scopes.include?("group")
+          ids = access_id_list(verb, klass)
+          !(klass.try(:supports_access_control?) && ids.exclude?(id.to_s))
+        elsif scopes.include?("user")
+          @record.owner == @user.user.user.username
+        else
+          false
```

Comment:
@eclarizio Can we add an error message here with the details of the class/id/verb so we can debug

Reply:
Will do